packages/rpc-methods/src/restricted/invokeSnap.ts

```diff
@@ -49,22 +82,18 @@ type InvokeSnapSpecification = ValidPermissionSpecification<{
  * and install it if it's not avaialble yet.
  *
  * @param options - The specification builder options.
- * @param options.allowedCaveats - The optional allowed caveats for the permission.
  * @param options.methodHooks - The RPC method hooks needed by the method implementation.
  * @returns The specification for the `wallet_snap_*` permission.
  */
 const specificationBuilder: PermissionSpecificationBuilder<
   PermissionType.RestrictedMethod,
   InvokeSnapSpecificationBuilderOptions,
   InvokeSnapSpecification
-> = ({
-  allowedCaveats = null,
-  methodHooks,
-}: InvokeSnapSpecificationBuilderOptions) => {
+> = ({ methodHooks }: InvokeSnapSpecificationBuilderOptions) => {
   return {
     permissionType: PermissionType.RestrictedMethod,
     targetKey,
-    allowedCaveats,
+    allowedCaveats: [SnapCaveatType.SnapIds],
     methodImplementation: getInvokeSnapImplementation(methodHooks),
```

Comment:
We should validate that the permission only has the one caveat we expect:  https://github.com/MetaMask/snaps-monorepo/blob/main/packages/rpc-methods/src/restricted/getBip44Entropy.ts#L123-L130

Reply:
This one doesn't look solved yet, unless it is pending a push?

Reply:
yes!